### ISSUE TITLE
Lack of simple yet complete examples.

### ISSUE THREAD

**Issue by kosaho** [leafo/lapis#522]
The framework does have indeed a good documentation, but... I mean, it's really hard to get used to it, even after reading the documentation. Even better than guides, simple examples would provide a much easier adaptation. For example, a simple blog working with sessions.

The mysql documentation does not show an exemple either, so is the db connection escaped? I mean, does it works like PDO prepare (verbatim)?

**Comment by Guard13007** [comment]
What kind of examples besides a blog would you want? I'm totally up for working on such examples as practice and to learn the framework better.

**Comment by kosaho** [comment]
Thank you. I would be grateful if you could write an simple login system using sessions, since I couldn't get it.
You don't really have to write a entire blog example, I'm just wondering how files are separated (best practices guideline). For example, app.lua + posts.lua + pages.lua (?).

**Comment by sliceofcode** [comment]
We could borrow some inspiration from Flask. The Flask tutorial has you build an example app from scratch. Perhaps a Lapis tutorial could do the same? The screencast that Leafo put up on their YouTube channel is good, but it's outdated by a few years.

**Comment by kosaho** [comment]
I second that, the Flask tutorial looks really interesting.

**Comment by honewatson** [comment]
Streak.club code base is good for examples.

**Comment by DarkWiiPlayer** [comment]
I second this. Lapis has good documentation for the basic stuff, but users are pretty much left to themselves with anything more complex than that, and figuring out all that stuff takes time.

**Comment by Guard13007** [comment]
@sliceofcode Can you link to that video? Maybe it'll give me an idea for what should be covered.

**Comment by CatLover91** [comment]
I agree with @honewatson that streak.club is a spectacular example. Streak.club would really benefit from code annotations like how [underscore does it](http://underscorejs.org/docs/underscore.html). Apparently it uses [docco](http://ashkenas.com/docco/) although for posterity's sake it would be more fitting that the lua-flavored [locco](http://rgieseke.github.io/locco/) is used. Would anyone be interested in an idea like this happening?

**Comment by honewatson** [comment]
@CatLover91 locco is cool though I also like this style of inline documentation using Markdown:

https://github.com/alexanderGugel/jdi

Shouldn't be too hard to create something for moonscript.  I have already started creating some comments in anticipation of creating something like jdi for moonscript at some stage.
```moonscript
-- ## Confirm Anonymous Variable
-- 
-- Check if anonymous value exists.  If it doesn't assign anonymous property with false
check_anonymous = (user) ->
  unless user.anonymous
    user.anonymous = false
  user
```

**Comment by CatLover91** [comment]
@honewatson I am willing to come up with some kind of hosting and a tld to stand up something like this. I think the lapis community would really benefit from more resources and it would help distribute some of the weight of providing documentation off @leafo's shoulders.
